driver/InputFlags.cpp

```diff
@@ -124,9 +127,30 @@ void InputFlags::Parse(int argc, char* argv[])
                 Print();
 
             char short_name = FindShortName(long_name);
-
-            MapInputs[short_name].value = args[i + 1];
-            i++;
+            
+            //check for short name of in layout flags
+            if ( (short_name == 'I') || (short_name == 'O') || (short_name == 'f'))
```

Comment:
This code implicitly depends on layout option definitions.

Reply:
I have added these short name checks to make sure, lib is looking at correct input strings. As these are supplied by user. String value "NHCW/NCHW" etc are w.r.t short name I/O/f which are related to in layout/out layout/fill layout flags.

Reply:
[Note] If you change "short name" of an option, then you'll need to change this code. But this dependence is implicit: if you forget to do that, then compilation failure will NOT happen; but the driver will work incorrectly. That is why I said that "this code implicitly depends on layout option definitions."

Actually I do not think we need these checks in the driver at all, because "The driver should not limit layouts. Only library knows which layouts are supported" (see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1316#discussion_r767329617)

Reply:
https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1316#discussion_r762338663 
short name checks are included, to accommodate the review comment.

Reply:
This is very true, but what I am asking for is total redesign of this feature, for the sake of not adding new implicit dependencies in our code (maybe at the cost of UX, but out users are engineers, right?)

Reply:
@technicalgrp89 
>> total redesign of this feature, for the sake of not adding new implicit dependencies in our code . 

> can you explain, what all should be redesigned?

This is very simple. This PR is intended to implement a new small feature mentioned in #1311, i.e. better handling of incorrect layouts. I think that the implementation of that feature (i.e. this PR) should be redesigned.

> our intension is to check layout flag value strings. so I did not understand, let me knw what is needed. thank you

The intention is good, but current implementation introduces dependencies between the driver and the library. If the library will be modified to support more (or less) layouts, then the driver needs to be modified as well. And this dependence is not expressed programmatically, which makes it _implicit_.

Another implicit dependence is mentioned [here](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1316#discussion_r768093501).

Each implicit dependence means a code quality issue. Does the improved diagnostics worth two more quality issues in the code? I suspect that this is questionable, and that is why I am asking for "total redesign" (which is described [here](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1316#discussion_r768160664)):

> ...I believe you can propose and implement much better error message in the library.

Maybe I am wrong, but who knows? @JehandadKhan what do you think?

Reply:
@JehandadKhan , let me knw . thank you

Reply:
@technicalgrp89 I agree with @atamazov. There should not be such dependency between the library and the driver and separation of function and responsibility should be ensured. Lets setup a meeting to go over your PR